runtime/oti/j9.h

```diff
@@ -370,9 +369,9 @@ static const struct { \
 #define J9_VALUETYPE_FLATTENED_SIZE(clazz)((UDATA) 0) /* It is not possible for this macro to be used since we always check J9_IS_J9CLASS_FLATTENED before ever using it. */
 #define J9_IS_J9ARRAYCLASS_NULL_RESTRICTED(clazz) FALSE
 #define J9CLASS_GET_NULLRESTRICTED_ARRAY(clazz) NULL
-#define J9ROMFIELD_IS_STRICT(romClassOrClassfile, fieldModifiers) FALSE
 #endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
 #define IS_CLASS_SIGNATURE(firstChar) ('L' == (firstChar))
+#define J9ROMFIELD_IS_STRICT(romClassOrClassfile, fieldModifiers) (J9_IS_CLASSFILE_OR_ROMCLASS_VALUETYPE_VERSION(romClassOrClassfile) && J9_ARE_ALL_BITS_SET(fieldModifiers, J9AccStrictInit))
```

Comment:
Please add a todo comment here to update` J9_IS_CLASSFILE_OR_ROMCLASS_VALUETYPE_VERSION()` if the strict field is released before Valhalla.